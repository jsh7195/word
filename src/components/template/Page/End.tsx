import React, { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";
import { meanBy , filter} from 'lodash';
import { WordsItem } from "@lib/data";
import Button from "@atoms/Button";
import Point from '@atoms/Point';

const End = () => {
  const location = useLocation();
  const [ point, setPoint ] = useState(0);
  const [ avrTime, setAvrTime ] = useState(0);

  useEffect(() => {
    if (location.state) {
      const { params, point }: any = location.state;
      setPoint(point);
      
      // 성공 평균시간 
      // 계산범위는 정수이므로 소요시간이 0.xx초인 문제는 올림해서 1초로 계산한다.
      const successList = filter(params,(item:WordsItem)=>{
        return item.success;
      })
      //평균 소수점 2자리 반올림
      const avg = meanBy(successList, (item:WordsItem) => {
        if(item.success){
          return item.spend === 0 ? 1 : item.spend;
        }
      }).toFixed(2);
      setAvrTime(Number(avg));

    }
  }, [location.state]);

  return (
    <div>
      <div>결과화면</div>
      <span>평균 소요시간 : </span>{avrTime} <br/>
      <Point point={point}/> <br/>
      <Button text="다시시작" url="/Intro" />
    </div>
  );
};

export default End;
