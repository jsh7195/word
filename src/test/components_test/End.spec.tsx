import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import End from "@template/Page/End";

Enzyme.configure({ adapter: new Adapter() });

test("End Render Test ... <End/>", () => {
  const component = renderer
    .create(
      <BrowserRouter>
        <End />
      </BrowserRouter>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
