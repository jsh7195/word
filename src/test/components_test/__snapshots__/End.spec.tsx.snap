// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End Render Test ... <End/> 1`] = `
<div>
  <div>
    결과화면
  </div>
  <span>
    평균 소요시간 : 
  </span>
  0
   
  <br />
  <span>
    점수 : 
    0
     점
  </span>
   
  <br />
  <button>
    <a
      href="/Intro"
      onClick={[Function]}
    >
      다시시작
    </a>
  </button>
</div>
`;
